pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -36,9 +36,35 @@ const (
 )
 
 var (
-	defaultReleaseImage = "registry.svc.ci.openshift.org/openshift/origin-release:v4.0"
+	// defaultReleaseImageOriginal is the value served when defaultReleaseImagePadded is unmodified.
+	defaultReleaseImageOriginal = "registry.svc.ci.openshift.org/openshift/origin-release:v4.0"
+	defaultReleaseImagePrefix   = ""
+	// defaultReleaseImagePadded may be replaced in the binary with a pull spec that overrides defaultReleaseImage as
+	// a null-terminated string within the allowed character length. This allows a distributor to override the payload
+	// location without having to rebuild the source.
+	defaultReleaseImagePadded = "_RELEASE_IMAGE_LOCATION_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
```

Comment:
Static strings in Go executables are not null-terminated by default:

```console
$ git describe
unreleased-master-564-g5886deac9
$ hack/build.sh
$ hexdump -C bin/openshift-install | grep -1 XXXXXXXXXXX
04af07f0  45 41 53 45 5f 49 4d 41  47 45 5f 4c 4f 43 41 54  |EASE_IMAGE_LOCAT|
04af0800  49 4f 4e 5f 58 58 58 58  58 58 58 58 58 58 58 58  |ION_XXXXXXXXXXXX|
04af0810  58 58 58 58 58 58 58 58  58 58 58 58 58 58 58 58  |XXXXXXXXXXXXXXXX|
*
--
04da9c30  7f aa aa aa aa aa aa aa  aa aa aa aa aa aa 28 2c  |..............(,|
04da9c40  2c 7f 58 58 58 58 58 58  58 58 58 58 58 58 58 58  |,.XXXXXXXXXXXXXX|
04da9c50  58 58 7f b0 b0 b0 b0 b0  b0 b0 b0 b0 b0 b0 b0 b0  |XX..............|
```

So I think we want to explicitly include a terminal `\x00` here.

Reply:
in compilation the length of the slice given to the code should be set to the length of the constant (based on my understanding of internals).

Maybe we should do `\x00_PREFIX[\x00...]` - the X was mostly arbitrary so as to look like underfill.  The string PREFIX could show up in the code elsewhere though.